frontend/src/pages/DetailedCourse/DetailedCourseContainer.tsx

```diff
@@ -24,6 +24,11 @@ const DetailedCourseContainer: React.FC<Props> = ({ page }) => {
   const { mutate, isLoading } = useApplyCourse();
   const [isFullTextOpen, setFullTextOpen] = useState(false);
   const isCourseApplicationSubmitted = courseData ? Boolean(courseData.status) : false;
+  const isProgressBarDisplayed = courseData
```

Comment:
might i ask you to bring out all logic connected with Progress bar to own "progress bar" component and make it reusable in all components where progress bar is

Reply:
It's ProgressBar conditional rendering only for Detailed Course page, so I think this logic should be here, not in ProgressBar container

Reply:
@k-krasilnikova could you please comment this issue?